Engine/Code/Engine/Graphics/MeshBuilder.h

```diff
@@ -40,35 +40,37 @@ struct VertexMaster
 	Vec4 m_Color;
 	Vec2 m_UV;
 
-	static int m_Size;
+	VertexMaster() {}
+	VertexMaster(Vec3 position, Vec4 color, Vec2 uv) : m_Pos(position), m_Color(color), m_UV(uv) {}
+	~VertexMaster() {}
+};
 
-	VertexMaster(Vec3 position, Vec4 color, Vec2 uv);
-	~VertexMaster();
+struct VertexPCU
+{
+	Vec3 m_Pos;
+	Vec4 m_Color;
+	Vec2 m_UV;
 
-	void PushPosCoords(Vec3 position);
-	void PushColorCoords(Vec4 color);
-	void PushUVCoords(Vec2 uv);
+	VertexPCU() {}
+	VertexPCU(VertexMaster master);
+	VertexPCU(Vec3 position, Vec4 color, Vec2 uv);
+	~VertexPCU();
 
-	struct PCU
-	{
-		static std::vector<VertexBufferElement> m_Attributes;
-		static VertexBufferLayout m_Layout;
-	};
+	static std::vector<VertexBufferElement> m_Attributes;
+	static VertexBufferLayout m_Layout;
 };
 
 struct MeshBuilder
 {
-	//std::vector<VertexPCU> m_Vertices;
 	std::vector<VertexMaster> m_Vertices;
```

Comment:
templatize this

Reply:
Lets not. Builder is supposed to be type agnostic